### PR TITLE
Add convenience `Content.withId(String)`

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

File: api/model/src/main/java/org/projectnessie/model/Content.java [hunk 1]
```diff
@@ -97,6 +97,8 @@ public interface Type {
   @JsonIgnore
   public abstract Type getType();
 
+  public abstract Content withId(String id);
```
Comment (adutra) [inline api/model/src/main/java/org/projectnessie/model/Content.java:100]: Maybe:

```suggestion
  public abstract Content withId(@Nullable @jakarta.annotation.Nullable String id);
```

Conversely we could require `id` to be non null and introduce a `withoutId()` method, wdyt?

File: versioned/storage/store/src/main/java/org/projectnessie/versioned/storage/versionstore/ContentMapping.java [hunk 2]
```diff
@@ -107,7 +107,7 @@ public ContentValueObj buildContent(
   @jakarta.annotation.Nonnull
   public Content assignContentId(
       @Nonnull @jakarta.annotation.Nonnull Content putValue, String contentId) {
-    return STORE_WORKER.applyId(putValue, contentId);
+    return putValue.withId(contentId);
```
Comment (adutra) [inline versioned/storage/store/src/main/java/org/projectnessie/versioned/storage/versionstore/ContentMapping.java:110]: Method `assignContentId` seems unused.

File: versioned/tests/src/main/java/org/projectnessie/versioned/testworker/OnRefOnly.java [hunk 3]
```diff
@@ -52,10 +52,9 @@ public ByteString serialized() {
   }
 
   public OnRefOnly withId(Content c) {
-    return onRef(getOnRef(), c.getId());
+    return withId(c.getId());
```
Comment (adutra) [inline versioned/tests/src/main/java/org/projectnessie/versioned/testworker/OnRefOnly.java:55]: Maybe inline?

File: api/model/src/main/java/org/projectnessie/model/CommitResponse.java [hunk 4]
```diff
@@ -61,6 +61,24 @@ default Map<ContentKey, String> toAddedContentsMap() {
...
+  default <T extends Content> T contentWithId(ContentKey key, T content) {
+    List<AddedContent> addedContents = getAddedContents();
+    if (addedContents == null) {
+      return content;
+    }
+    return (T)
+        getAddedContents().stream()
+            .filter(added -> added.getKey().equals(key))
+            .findFirst()
+            .map(added -> content.withId(added.contentId()))
```
Comment (dimas-b) [inline api/model/src/main/java/org/projectnessie/model/CommitResponse.java:78]: Not sure I understand the purpose of this method :thinking: why not simply return `added` to the caller?

File: api/model/src/main/java/org/projectnessie/model/CommitResponse.java [hunk 5]
```diff
@@ -61,6 +61,24 @@ default Map<ContentKey, String> toAddedContentsMap() {
     return added.stream().collect(Collectors.toMap(AddedContent::getKey, AddedContent::contentId));
   }
 
+  /**
+   * If new content has been added to Nessie, updates the {@link Content#getId() content ID} of the
+   * given content with the generated content ID.
+   */
+  @SuppressWarnings("unchecked")
+  default <T extends Content> T contentWithId(ContentKey key, T content) {
```
Comment (adutra) [inline api/model/src/main/java/org/projectnessie/model/CommitResponse.java:69]: Seems unused.

File: api/model/src/main/java/org/projectnessie/model/Content.java (hunk 1 above)
Comment (snazy) [reply api/model/src/main/java/org/projectnessie/model/Content.java:100]: Nope - ID must be nullable - otherwise you could never add new content

File: api/model/src/main/java/org/projectnessie/model/CommitResponse.java (hunk 4 above)
Comment (dimas-b) [reply api/model/src/main/java/org/projectnessie/model/CommitResponse.java:78]: right :facepalm:

File: api/model/src/main/java/org/projectnessie/model/CommitResponse.java [hunk 6]
```diff
@@ -61,6 +61,24 @@ default Map<ContentKey, String> toAddedContentsMap() {
...
+    List<AddedContent> addedContents = getAddedContents();
+    if (addedContents == null) {
+      return content;
```
Comment (dimas-b) [inline api/model/src/main/java/org/projectnessie/model/CommitResponse.java:72]: If the purpose of this method is for clients to obtain content IDs that have just been assigned, should we throw an exception here? If we get here that means the caller passed in a content that was either not part of the commit request, or was not "added". WDYT?

Comment (dimas-b) [reply api/model/src/main/java/org/projectnessie/model/CommitResponse.java:72]: well, but why are we adding this method? Like @adutra mentioned, it does not appear to be used ATM :thinking:

**Review by dimas-b — APPROVED** [review]
